From 7ccf084b7a281d8aa877f91401d3df7192da408c Mon Sep 17 00:00:00 2001
From: Chris Wilson <chris@chris-wilson.co.uk>
Date: Mon, 12 Oct 2009 14:33:08 +0100
Subject: [PATCH 2/3] uxa: Free the ScratchPixmapHeader after its associated Picture

Fixes: http://bugs.freedesktop.org/show_bug.cgi?id=24459
Intel Driver > 2.8: Cairo rendering bug, triggered in QtCurve GTK engine

Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>
(cherry picked from commit 7e8f32d0a7279dce1976f87612833d9092554cfe)
---
 uxa/uxa-render.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/uxa/uxa-render.c b/uxa/uxa-render.c
index 4850185..bf1460f 100644
--- a/uxa/uxa-render.c
+++ b/uxa/uxa-render.c
@@ -360,8 +360,8 @@ uxa_picture_from_a1_pixman_image (ScreenPtr pScreen,
     pSrc = CreatePicture (0, &pPixmap->drawable,
 			  PictureMatchFormat (pScreen, 1, PICT_a1),
 			  0, 0, serverClient, &error);
-    FreeScratchPixmapHeader (pPixmap);
     if (!pSrc) {
+	FreeScratchPixmapHeader (pPixmap);
 	FreePicture (pPicture, 0);
 	return 0;
     }
@@ -373,6 +373,7 @@ uxa_picture_from_a1_pixman_image (ScreenPtr pScreen,
 		      width, height);
 
     FreePicture (pSrc, 0);
+    FreeScratchPixmapHeader (pPixmap);
 
     return pPicture;
 }
-- 
1.6.5.1

